src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/ErrorArchiverService.java

```diff
@@ -0,0 +1,119 @@
+package it.gov.pagopa.payhub.activities.service.ingestionflow;
+
+import it.gov.pagopa.payhub.activities.dto.ingestion.IngestionFlowFileErrorDTO;
+import it.gov.pagopa.payhub.activities.exception.NotRetryableActivityException;
+import it.gov.pagopa.payhub.activities.service.CsvService;
+import it.gov.pagopa.payhub.activities.util.Utilities;
+import it.gov.pagopa.pu.processexecutions.dto.generated.IngestionFlowFile;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+
+@Slf4j
+public abstract class ErrorArchiverService<T extends IngestionFlowFileErrorDTO> {
+
+    private static final String ERRORFILE_PREFIX = "ERROR-";
+
+    private final Path sharedDirectoryPath;
+    private final String errorFolder;
+    private final IngestionFlowFileArchiverService ingestionFlowFileArchiverService;
+    private final CsvService csvService;
+
+    protected ErrorArchiverService(
+            String sharedFolder,
+            String errorFolder,
+            IngestionFlowFileArchiverService ingestionFlowFileArchiverService,
+            CsvService csvService
+    ) {
+        this.sharedDirectoryPath = Path.of(sharedFolder);
+        this.errorFolder = errorFolder;
+        this.ingestionFlowFileArchiverService = ingestionFlowFileArchiverService;
+        this.csvService = csvService;
+    }
+
+    /**
+     * Writes error data into a CSV file.
+     *
+     * @param workingDirectory  The directory where the file should be created.
+     * @param ingestionFlowFile The metadata of the ingestion file.
+     * @param errorList         The list of errors to write.
+     * @param headers           The headers of the CSV file.
+     */
+    public void writeErrors(Path workingDirectory, IngestionFlowFile ingestionFlowFile,
+                            List<T> errorList, List<String> headers) {
+
+        if(errorList.isEmpty()){
```

Comment:
Maybe it could be null?
```suggestion
        if(CollectionUtils.isEmpty(errorList)){
```

Reply:
@domenicogi